### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by YuunoKun**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -241,6 +244,42 @@ _{more aspects and alternatives to be added}_
 _{Explain here how the data archiving feature will be implemented}_
 
 
+### <a name="adding_a_session"></a>Adding a session
+
+Adding a session to a class requires user input from the CLI, then the parser will parse the user input and 
+to obtain the name and the date of the session. The newly added session will also be filled with students'
```
Comment (nweiyue) [inline docs/DeveloperGuide.md]: Would "Adding a session to a class requires user input from the CLI. The parser will then parse the user input to obtain the name and the date of the session." flow better?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -241,6 +244,42 @@ _{more aspects and alternatives to be added}_
...
+Adding a session to a class requires user input from the CLI, then the parser will parse the user input and 
+to obtain the name and the date of the session. The newly added session will also be filled with students' 
+default presence and participation.
+
+The below example given assumes that the user inputs `addses s/Tutorial 1 d/1/1/2020`, which 'Tutorial 1' and
+'1/1/2020' are the name and the date of the session to be added, and `addses` command has already been parsed.
+
+Step 1: Parse input and create session
+
+A `AddSessionCommandParser` is created by `LogicManager` and calls the parse method with the given
```
Comment (nweiyue) [inline docs/DeveloperGuide.md]: * Should this be "An"? I've noticed this in other places as well.
* Perhaps you can use markdown `AddSessionCommand#parse()` to replace "parse method'? I've noticed this in line 272 and 277 for other methods as well.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -241,6 +244,42 @@ _{more aspects and alternatives to be added}_
...
+user input. Input is parsed according to the prefix `s/` and `d/` to identify the parts of the 
+user input and split it into String slices. If both the name and date are present in the input,
```
Comment (nweiyue) [inline docs/DeveloperGuide.md]: Perhaps "String" can `String`? I've noticed this in other places as well.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -241,6 +244,42 @@ _{more aspects and alternatives to be added}_
...
+![AddSessionSequenceDiagram1](images/AddSessionSequenceDiagram1.png)
+
+Step 2: Add session to the model/session list
+
+`LogicManager` calls execute method to proceed on to adding the session to the model.
+The `AddSessionCommand` will first check if the same session is already in the session list, if the 
+session exists in the current session list, then the session will not be added to the session list.
+
+Assuming that the session to add is not a session already in the list, `ModelManager` will update
+the internal student list of the session list, then it uses addSession method to add the session to the
```
Comment (nweiyue) [inline docs/DeveloperGuide.md:439]: Perhaps breaking this paragraph into two sentences will make it flow better? For example: "... . It then uses `ModelManager#addSession()` .."

**Review by nweiyue — APPROVED** [review]
LGTM! Nice diagrams! Maybe just some suggestions to make your explanation clearer.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (YuunoKun) [reply docs/DeveloperGuide.md]: I think I deleted some stuff by accident, will amend!

**Pull request merged**